src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -4033,11 +4033,90 @@ uint64_t BlockchainLMDB::get_database_size() const
   return size;
 }
 
-void BlockchainLMDB::fixup()
+void BlockchainLMDB::fixup(fixup_context context)
 {
   LOG_PRINT_L3("BlockchainLMDB::" << __func__);
   // Always call parent as well
-  BlockchainDB::fixup();
+  BlockchainDB::fixup(context);
+
+  if (is_read_only())
+    return;
+
+  if (context.type == fixup_type::calculate_difficulty)
+  {
+    if (context.calculate_difficulty.start_height == 0)
+      context.calculate_difficulty.start_height++;
+
+    uint64_t const start_height = context.calculate_difficulty.start_height;
```

Comment:
do check for 0 and assignment of the local here using ternary operator:

`uint64_t const start_height = context.calculate_difficulty.start_height == 0 ? 1 : context.calculate_difficulty.start_height;`

Reply:
Fixed